lib/web/apiserver.go

```diff
@@ -2254,6 +2305,52 @@ func (h *Handler) siteNodeConnect(
 	return nil, nil
 }
 
+func (h *Handler) generateSession(ctx context.Context, clt auth.ClientI, req *TerminalRequest, clusterName string) (session.Session, error) {
+	var (
+		host string
+		port int
+	)
+
+	// req.Server will be either an IP, hostname, or server UUID.
+	_, err := uuid.Parse(req.Server)
+	if err != nil {
+		servers, err := clt.GetNodes(ctx, apidefaults.Namespace)
```

Comment:
Don't let this block you, but maybe worth leaving a TODO.

This is the "slow path" - right? We have to filter through all nodes to find the right one. If that's the case, would using `ListResources` instead of `GetNodes` make more sense here? With `ListResources` I think we can supply more filters in the query, which might result in less work to do client-side.

Reply:
Todo added.